platform/darwin/src/MGLGeoJSONSource.mm

```diff
@@ -82,9 +93,8 @@ - (void)commonInit
         _features = nil;
     } else if (self.geoJSONData) {
         NSString *string = [[NSString alloc] initWithData:self.geoJSONData encoding:NSUTF8StringEncoding];
-        const auto geojson = mapbox::geojson::parse(string.UTF8String).get<mapbox::geojson::feature_collection>();
+        const auto geojson = mapbox::geojson::parse(string.UTF8String);
         source->setGeoJSON(geojson);
-        _features = MGLFeaturesFromMBGLFeatures(geojson);
```

Comment:
If we're gonna support plain geometries, we can't assign features here.
Wouldn't it make sense to rename features to geometries and write a GeoJSON evaluator which overloads  featureCollection, feature and geometry?